simplipy/system/__init__.py

```diff
@@ -343,6 +343,15 @@ def generate_device_objects(self) -> None:
         """
         raise NotImplementedError()
 
+    async def async_get_media(self, url: str) -> bytes:
+        """Return a media file in bytes.
+
+        Outdoor camera motion events (for example) generate media urls that do not have a common base
+        like the rest of the SS apis, but do require the same bearer token to access.  Use this method to
+        fetch them.
+        """
+        return self._api.async_request("get", url)
```

Comment:
Feels like this should do more of it's going to be its own method. Why not just call the request method directly when needed (like you do here)?

Reply:
If you mean HA's Simplisafe/camera.py should call something like self.system._api.async_request(), I thought _api was "private" and shouldn't be called "outside".  I was getting linter errors when I was doing that earlier.  Or am I misunderstanding your issue with this?

Reply:
Maybe I'm misunderstanding—do you envision implementers of this library calling this method on demand?

Reply:
Oops, sorry that is a mistake.  Left over from older attempts.  I will remove it.  The public api is in api.py; async_media_request(url: str) -> bytes | None

This public api will be called by consumers (like HA) to fetch media urls.

Reply:
I removed that code and pushed.